### ISSUE TITLE
AWS STS session tags

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#10994]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

You can now add key-value pair attributes when you assume an IAM role using AWS STS.

### New or Affected Resource(s)

The [AWS provider](https://www.terraform.io/docs/providers/aws/index.html) itself.

### Potential Terraform Configuration

```hcl
provider "aws" {
  assume_role {
    role_arn     = "arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME"
    session_name = "SESSION_NAME"
    external_id  = "EXTERNAL_ID"

    tags = {
      Usage = "Terraform"
    }
  }
}
```

### References

[User guide](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_session-tags.html).

Requires [AWS SDK v1.25.41](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.41):
* https://github.com/terraform-providers/terraform-provider-aws/pull/10955

**Comment by ewbankkit** [comment]
May require enhancements to [`aws-sdk-go-base`](https://github.com/hashicorp/aws-sdk-go-base).

**Comment by ewbankkit** [comment]
Related:
* https://github.com/aws/aws-sdk-go/pull/2993

Requires [AWS SDK v1.26.0](https://github.com/aws/aws-sdk-go/releases/tag/v1.26.0):
* https://github.com/terraform-providers/terraform-provider-aws/pull/11254

**Comment by robh007** [comment]
I think this requires an update to [https://github.com/hashicorp/aws-sdk-go-base](aws-sdk-go).

**Comment by ewbankkit** [comment]
Closed via https://github.com/hashicorp/aws-sdk-go-base/pull/39.
Functionality should be available in AWS Provider v3.0.0.
